### ISSUE TITLE
UserWarning with num_iterations

### ISSUE THREAD

**Issue by JAroyan** [microsoft/LightGBM#6385]
A UserWarning consistently appears: UserWarning: Found 'n_estimators' in params. Will use it instead of argument.

According to the documentation, num_iterations is the parameter, with the following aliases: num_iteration, n_iter, num_tree, num_trees, num_round, num_rounds, nrounds, num_boost_round, n_estimators, max_iter.

When I change the parameter to num_iterations, this warning should not appear.

However, the warning now states: UserWarning: Found 'num_iterations' in params. Will use it instead of argument.
_log_warning(f"Found '{alias}' in params. Will use it instead of argument")

The bug can be in engine.py within the train function:

    for alias in _ConfigAliases.get("num_iterations"):
        all_aliases = _ConfigAliases.get("num_iterations") ### I debugged this, and the set contains also num_iterations. I think without this string the warning would not occur.
        if alias in params:
            num_boost_round = params.pop(alias)
            _log_warning(f"Found `{alias}` in params. Will use it instead of argument")

With this modification the warning occurs only if an alias was used.

    for alias in _ConfigAliases.get("num_iterations"):
        all_aliases = _ConfigAliases.get("num_iterations")
        if alias == "num_iterations":
            continue
        if alias in params:
            num_boost_round = params.pop(alias)
            _log_warning(f"Found `{alias}` in params. Will use it instead of argument")

My params:
lightgbm:
    objective: "regression"
    metric: "rmse"
    boosting: "gbdt" 
    lambda_l1: 0.01
    lambda_l2: 0.01
    num_leaves: 16
    max_depth: -1
    num_iterations: 2000
    learning_rate: 0.01
    early_stopping_rounds: 50
    verbosity: 1 

LightGBM Version: 4.3.0

**Comment by YingJie-Zhao** [comment]
Did you set 'num_iterations' to 'params' and 'n_estimators' to function call parameters at the same time?

e.g.
```python
param = {
    'num_iterations': 100
}

model = lgb.train(params, n_estimators=50)
```

**Comment by JAroyan** [comment]
No, i dont use num_iterations and aliases at the same time.

**Comment by jameslamb** [comment]
Thanks for using LightGBM.

This is already being discussed #6324, please subscribe to that issue for updates. You can also find more details on strategies for suppressing that warning at https://stackoverflow.com/a/78213401/3986677.

For the future, you might also find that these are helpful for working on GitHub:

* how to format code with markdown: https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax
* how to link to code: https://docs.github.com/en/repositories/working-with-files/using-files/getting-permanent-links-to-files
